dnf5/commands/history/history_store.cpp

```diff
@@ -19,14 +19,75 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 #include "history_store.hpp"
 
+#include "commands/history/transaction_id.hpp"
+
+#include <libdnf5/utils/bgettext/bgettext-mark-domain.h>
+
+#include <fstream>
+
 namespace dnf5 {
 
 using namespace libdnf5::cli;
 
 void HistoryStoreCommand::set_argument_parser() {
-    get_argument_parser_command()->set_description("Store transaction to a file");
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_description("[experimental] Store transaction to a file");
+    auto & ctx = get_context();
+    auto & parser = ctx.get_argument_parser();
+
+    output_option = dynamic_cast<libdnf5::OptionString *>(
+        parser.add_init_value(std::make_unique<libdnf5::OptionString>("./transaction.json")));
+    auto query_format = parser.add_new_named_arg("output");
+    query_format->set_long_name("output");
+    query_format->set_short_name('o');
+    query_format->set_description("File path for storing the transaction, default is \"./transaction.json\"");
+    query_format->set_has_value(true);
+    query_format->set_arg_value_help("PATH");
+    query_format->link_value(output_option);
+    cmd.register_named_arg(query_format);
+
+    transaction_specs = std::make_unique<TransactionSpecArguments>(*this);
 }
 
-void HistoryStoreCommand::run() {}
+void HistoryStoreCommand::run() {
+    const auto ts_specs = transaction_specs->get_value();
+    auto & history = *get_context().base.get_transaction_history();
+    std::vector<libdnf5::transaction::Transaction> transactions;
+
+    if (ts_specs.empty()) {
+        transactions = list_transactions_from_specs(history, {"last"});
+    } else {
+        transactions = list_transactions_from_specs(history, ts_specs);
+    }
+
+    //TODO(amatej): Allow storing multiple transactions at once?
+    if (transactions.size() != 1) {
+        throw libdnf5::cli::CommandExitError(1, M_("Multiple transactions selected for storing, only one allowed."));
+    }
+
+    const std::string json = transactions[0].serialize_to_json();
+
+    //TODO(amatej): maybe we could expose libdnf5/utils/fs/temp.hpp and use it here
+    std::filesystem::path tmp_path(output_option->get_value() + ".XXXXXX");
+    std::string dest = tmp_path;
+    int fd = mkstemp(dest.data());
+    if (fd == -1) {
+        throw std::filesystem::filesystem_error(
+            "cannot create temporary file", dest, std::error_code(errno, std::system_category()));
+    }
+
+    if (write(fd, json.data(), json.size()) == -1) {
+        throw std::filesystem::filesystem_error(
+            "cannot write to temporary file", dest, std::error_code(errno, std::system_category()));
+    }
```

Comment:
I believe that the code will keep open the file descriptor and also the temp file will remain on the system. What about following code?
```
close(fd);
std::error_code ec;
std::filesystem::remove(tmpfilepath, ec);
```

Reply:
I think it would be best to use the `libdnf5::utils::fs::temp.hpp` to handle this. I made a PR to expose it: https://github.com/rpm-software-management/dnf5/pull/1050